src/arrayMethodSort.js

```diff
@@ -5,7 +5,35 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    const defaultSort = (elementA, elemenrB) => {
+      return String(elementA) > String(elemenrB)
+        ? 1
+        : String(elementA) === String(elemenrB)
+          ? 0
+          : -1;
+    };
+
+    const currentSort = compareFunction === undefined
+      ? defaultSort
+      : compareFunction;
+
+    let count = 0;
+
+    do {
+      count = 0;
+
+      for (let i = 1; i < this.length; i++) {
+        if (currentSort(this[i - 1], this[i]) > 0) {
+          const transferValue = this[i - 1];
+
+          this[i - 1] = this[i];
+          this[i] = transferValue;
```

Comment:
You can use destructurization for swapping two elements values without adding an extra variable.